### PR TITLE
Fix for #2518

### LINKED ISSUE #2518: Bulk version of giveBlock
We have a `search` console command that is great at finding stuff related to a particular topic, like "cake" with the CakeLie module enabled. However, you then have to type out or copy paste each block if you want to eat *all* the cake!

Probably it wouldn't be hard to combine the two, create a `bulkGiveBlock` command that takes a search term, feeds that to `search`, filters out only the block hits, then executes `giveBlock` once for each search result, offering each output to the player's inventory (until inventory is full if there are too many hits)

Potential extra would be also having a version for `giveItem` or maybe find a way to complete #2293 first then add a single bulk variant. Also potentially related: #2430, #2293

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/38231872-bulk-version-of-giveblock?utm_campaign=plugin&utm_content=tracker%2F5279&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F5279&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by nihal111**
### Contains

Fixes #2518

### How to test

- Open Console
- type `bulkGiveBlock cake` to get all cake blocks :smile: 
- type `bulkGiveBlock core` to get a larger list and see what happens when inventory gets full.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/world/block/entity/BlockCommands.java [hunk 1]
```diff
@@ -275,18 +307,18 @@ public String giveBlock(
                 throw new IllegalArgumentException("No block found for '" + uri + "'");
             }
         } else if (matchingUris.isEmpty()) {
-        	return suggestItemIfAvailable(uri);
+            return suggestItemIfAvailable(uri);
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/world/block/entity/BlockCommands.java]: Nice catch

Comment (nihal111) [inline engine/src/main/java/org/terasology/world/block/entity/BlockCommands.java]: Autoformatting ftw :smile:

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
:cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake:

**Comment by Cervator** [comment]
Just for reference I added `bulkGiveBlocks` to the docs for better exposure. Best to be open about how to get more cake,faster!

File: engine/src/main/java/org/terasology/world/block/entity/BlockCommands.java [hunk 2]
```diff
@@ -233,6 +234,37 @@ public void replaceBlock(
...
+            @Sender EntityRef sender,
+            @CommandParam("searched") String searched) {
+        String searchLowercase = searched.toLowerCase();
+        List<String> blocks = findBlockMatches(searchLowercase);
+        String result = "Found " + blocks.size() + " block matches when searching for '" + searched + "'.";
+        if (blocks.size() > 0) {
+            result += "\nBlocks:";
+            for (String block : blocks) {
+                result += "\n" + block;
+                giveBlock(sender, block, null, null);
```
Comment (nihal111) [inline engine/src/main/java/org/terasology/world/block/entity/BlockCommands.java]: @Cervator, I could easily change this to allow a numberOfBlocks user input and give those many per block. Do we want that?

Comment (Cervator) [inline engine/src/main/java/org/terasology/world/block/entity/BlockCommands.java]: I would want *both* just like with giveBlocks. Just have a variant that takes a second parameter indicating count :-)

You could even add support for the shape option, but that's going a little far. Bulk amounts of staircase-shaped cake!

Comment (nihal111) [inline engine/src/main/java/org/terasology/world/block/entity/BlockCommands.java]: I don't know how the shape option works. Could you help me with that?

Comment (Cervator) [inline engine/src/main/java/org/terasology/world/block/entity/BlockCommands.java]: I think it is probably overkill, but that's fourth parameter right there, the second null. You can check out how it works by simply playing with giveBlock :-)

`giveblock stone 99 stair`

Should give you 99 stone stair-shaped blocks and is an example from the Playing doc ... although after fixing it as the doc has the quantity and shape switched ;-) Fixing next time I push something.

**Pull request merged**